### PR TITLE
catching JSONDecodeError

### PR DESCRIPTION
**Pull request by richardtief**
skip vm_id in VMPropertiesCollector

### CODE REVIEW

File: tools/Resources.py [hunk 1]
```diff
@@ -88,7 +88,7 @@ def get_resources(self, target, token, resourcekind, parentid):
                                     headers=headers)
         except Exception as e:
             print("Problem connecting to " + target + "Error: " + str(e))
-            return False
+            return []
```
Comment (viennaa) [inline tools/Resources.py]: That was long overdue :D

File: tools/Resources.py [hunk 2]
```diff
@@ -180,7 +180,12 @@ def get_latest_stat_multiple(target, token, uuids, key):
             return False
 
         if response.status_code == 200:
-            return response.json()['values']
+            try:
+                return response.json()['values']
+            except json.decoder.JSONDecodeError as e:
+                print("Catching JSONDecodeError for target:", str(target), "and key:", str(key),
```
Comment (viennaa) [inline tools/Resources.py:186]: please add spaces after the `:`

File: tools/Resources.py [hunk 3]
```diff
@@ -240,24 +245,29 @@ def get_latest_number_properties_multiple(target, token, uuids, propkey):
...
-            for resource in response.json()['values']:
-                d = dict()
-                d['resourceId'] = resource['resourceId']
-                d['propkey'] = propkey
-                content = resource['property-contents']['property-content']
-                if content:
-                    if 'values' in content[0]:
-                        d['data'] = content[0]['values'][0]
+            try:
+                if not response.json()['values']:
```
Comment (viennaa) [inline tools/Resources.py:249]: I'd say we only try catch this

File: tools/Resources.py [hunk 4]
```diff
@@ -294,28 +304,32 @@ def get_latest_enum_properties_multiple(target, token, uuids, propkey, expected_
...
-                content = resource['property-contents']['property-content']
-                if content:
-                    if 'values' in content[0]:
-                        latest_state = content[0]['values'][0]
-                    else:
-                        latest_state = "unknown"
-                    if latest_state == expected_state:
-                        d['data'] = 1
+            try:
+                if not response.json()['values']:
```
Comment (viennaa) [inline tools/Resources.py:308]: same here

File: tools/Resources.py [hunk 5]
```diff
@@ -351,24 +365,28 @@ def get_latest_info_properties_multiple(target, token, uuids, propkey):
...
-            for resource in response.json()['values']:
-                d = dict()
-                d['resourceId'] = resource['resourceId']
-                d['propkey'] = propkey
-                content = resource['property-contents']['property-content']
-                if content:
-                    if 'values' in content[0]:
-                        info = content[0]['values'][0]
+            try:
+                if not response.json()['values']:
```
Comment (viennaa) [inline tools/Resources.py:369]: and here

File: tools/Resources.py (hunk 2 above)
Comment (richardtief) [reply tools/Resources.py:186]: with a comma a space is inserted automatically

File: tools/Resources.py (hunk 3 above)
Comment (richardtief) [reply tools/Resources.py:249]: without catching JSONDecodeError and message?

Comment (viennaa) [reply tools/Resources.py:249]: The problem always occurs here with `['values']`, so why take in the other stuff too?

Comment (richardtief) [reply tools/Resources.py:249]: got it

**Pull request merged**